src/universal/utils/isObject.js

```diff
@@ -1,3 +1,3 @@
 export default function isObject(val) {
-  return val && typeof val === 'object';
+  return val && typeof val === 'object' && !(val instanceof Date);
```

Comment:
I draw the line at `null` and `Date` I promise :smile:
I think you're right though, let's roll this back. It doesn't impact anything and I can come up with something cleaner than this.